### ISSUE TITLE
Heroku

### ISSUE THREAD

**Issue by texmorgan** [GeekdomSA/bookclub#22]
http://rapgenius.com/James-somers-herokus-ugly-secret-lyrics

**Comment by warmwaffles** [comment]
Eh, this is just a bunch of devs that aren't smart enough to go get some actual server gear.

Anyways, the article is worth reading.

**Comment by richardjortega** [comment]
I think actually, it's a bunch of devs that have got hit by the HSF! (Heroku Swindle Factor)

I enjoy serverless deployment, and prefer it over dealing with sys admin work.

**Comment by warmwaffles** [comment]
> I enjoy serverless deployment, and prefer it over dealing with sys admin work.

This is true, however learning a little bit of Chef and how your system works, can go a long way in saving you money and time.

**Comment by wwkeyboard** [comment]
It's a bunch of people who don't understand load balancing and how your application fits into it. The crappy part is that Heroku changed from connection-oriented to round robin without mentioning it. Then they set a really short timeout on incoming requests(30 seconds, most browsers seem to allow 60) and start the counter when you hit their router. Because they start this at the routing layer when a requests actually reaches your process it might have very little time left before they return an error and bounce your dyno.

**Comment by warmwaffles** [comment]
Thus we have class action lawsuit against them now.

**Comment by gmcerveny** [comment]
This discussion makes me feel like it would be a good topic for discussion.
